app/src/main/java/com/example/development/sakaiclientandroid/utils/DataHandler.java

```diff
@@ -147,9 +156,12 @@ public void onResponse(Call<GradebookCollectionObject> call, Response<GradebookC
                 if(gradebookCollectionObject != null) {
                     Course currCourse = DataHandler.getCourseFromId(siteId);
                     currCourse.setGradebookObjectList(gradebookCollectionObject.getAssignments());
+                    UICallback.onSiteGradesSuccess(currCourse);
+                }
+                else {
+                    //TODO better error handling
+                    UICallback.onSiteGradesSuccess(null);
```

Comment:
@akench This should not be calling success if the request was unsuccessful (kind of, since no results were found). Make a request to `UICallback.onSiteGradesFailure()` and handle it properly on the frontend

Reply:
I was under the impression that the onSucess methods were for when the requests were a success, and same with onFailure. A site not having grades isn't a problem with the request. Also we want to show two seperate messages for when there aren't grades available ("No grades found for this course"), and for when the request fails ("There was a network error. please try again later", or something like that).